src/v/model/transform.cc

```diff
@@ -190,6 +190,27 @@ void transform_metadata::serde_write(iobuf& out) const {
     serde::write(out, compression_mode);
 }
 
+void transform_metadata::serde_read(iobuf_parser& in, const serde::header& h) {
+    using serde::read_nested;
+
+    name = read_nested<decltype(name)>(in, h._bytes_left_limit);
+    input_topic = read_nested<decltype(input_topic)>(in, h._bytes_left_limit);
+    output_topics = read_nested<decltype(output_topics)>(
+      in, h._bytes_left_limit);
+    environment = read_nested<decltype(environment)>(in, h._bytes_left_limit);
+    uuid = read_nested<decltype(uuid)>(in, h._bytes_left_limit);
+    source_ptr = read_nested<decltype(source_ptr)>(in, h._bytes_left_limit);
+    offset_options = read_nested<decltype(offset_options)>(
+      in, h._bytes_left_limit);
```

Comment:
cc @rockwotj i think i'm still a bit confused about how exactly std::variant serde works, but on the surface this looks like a correct transformation from serde_fields to serde_read.

Reply:
tangential, but I wonder whether we need the `bytes_left` checks in here...

I was using [topic_configuration::serde_read](https://github.com/redpanda-data/redpanda/blob/ac84d0dd9ecc490e6355689617016ecb0f94e1d1/src/v/cluster/topic_configuration.h#L82) as an example, but [column_store::serde_read](https://github.com/redpanda-data/redpanda/blob/ac84d0dd9ecc490e6355689617016ecb0f94e1d1/src/v/cloud_storage/segment_meta_cstore.cc#L736) is much more involved.

Reply:
i don't think you need this part

```cpp
   if (in.bytes_left() > h._bytes_left_limit) {
        in.skip(in.bytes_left() - h._bytes_left_limit);
    }
```

because it's done automatically in tag_invoke(read_tag) after serde_read is called. did you have a specific reason for adding it?

Reply:
cc @oleiman

Reply:
>done automatically inn `tag_invoke`

Ah, I misread that as embedded in the `!has_serde_read` branch.

>specific reason for adding it

Nope. Saw it written that way somewhere and it stuck for whatever reason.